### ISSUE TITLE
Cannot proceed to testing because metamask version folders/files stuck on setting up

### ISSUE THREAD

**Issue by Mouse-Coach** [TenKeyLabs/dappwright#406]
**Describe the bug**
When you try to run tests in parallel mode, sometimes tests are stucked because of "fs" error (attached in the "Logs" section). I guess it happens because of an incorrect "if" statement to delete the folder with "extractDestination".
<img width="790" alt="Screenshot 2024-12-23 at 15 12 27" src="https://github.com/user-attachments/assets/cd22162e-fc72-47fa-ac60-156345df050b" />

**To Reproduce**

1. Setup tests with metamask folder
2. Run several tests in parallel mode

**Logs**
```shell
Error: EEXIST: file already exists, mkdir '/tmp/dappwright/metamask/12_8_1'
```

**Expected behavior**
extractDestination folder successfully deleted - tests are not blocked

**Screenshots**
extractDestination folder is deleted - tests are blocked

**System:**

- OS: macOS
- OS version 12.4
- NodeJs version: 18.20.0
- dAppwright version: 2.9.1
- testing framework: Playwright
- testing framework version: 1.44.1

**Additional context**

**Comment by osis** [comment]
Ah yes, the current implementation cannot handle running tests asynchronously. There is an issue where it's discussed a bit. https://github.com/TenKeyLabs/dappwright/issues/3

**Comment by Mouse-Coach** [comment]
Hey @osis , besides that, there is another issue with calling the "bootstrap" method. 
It can't open the "Advanced" settings.
![Screenshot 2025-01-11 at 15 33 59](https://github.com/user-attachments/assets/a8eb6bb4-2fbc-445a-ac1a-f2bafca540ff)
<img width="633" alt="Screenshot 2025-01-11 at 15 33 38" src="https://github.com/user-attachments/assets/1f46472b-d691-407d-8209-181f2b099674" />

**Comment by Mouse-Coach** [comment]
@osis It's 2.9.1 (latest).

**Comment by osis** [comment]
Hmmm interesting. Do you by chance have a screenshot of where it's getting stuck?

**Comment by osis** [comment]
hey @Mouse-Coach, I ran into this as well updating all the tests to run in parallel. 

This is due to all the workers trying to download the library at the same time. I think the eventual fix would be that downloading should be restricted to worker 0, and the others should wait until its done. Not too hard to adjust but the code just isn't setup that way at the moment.

For now, what you can do is set `workers: 1` to make sure there is only one worker for that responsibility. Or, just run the lib a second time and the download should be complete.
